server/core/runtime/plan_step_runner.go

```diff
@@ -131,7 +131,7 @@ func (p *PlanStepRunner) switchWorkspace(ctx context.Context, prjCtx command.Pro
 			return err
 		}
 		// If `show` says we're already on this workspace then we're done.
-		if strings.TrimSpace(workspaceShowOutput) == prjCtx.Workspace {
+		if strings.Contains(workspaceShowOutput, prjCtx.Workspace) {
```

Comment:
Needed to change this line to a `Contains` check because the addition of trace logs around the output always made `strings.TrimSpace(workspaceShowOutput) == prjCtx.Workspace` return `False`.

Only thing I can think of is output the log into some separate file so that we don't print results of operations in the same place we print logs. It goes against the original plans for this feature, but I'm open to suggestions on other ideas!